src/typ.rs

```diff
@@ -38,6 +38,13 @@ pub enum Type {
 }
 
 impl Type {
+    pub fn is_result(&self) -> bool {
+        match self {
+            Type::App { name, .. } if &*name == "Result" => true,
```

Comment:
A style thing: Can we swap `&*name` for `name.as_str()`? I find it a little easier to read.

Also can we check that the module `is_empty`? In case someone implemented a type called Result in some other module.